src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -597,6 +597,7 @@ def __init__(
         interrupt_controller: Optional['InterruptController'] = None,  # G2: Cooperative cancellation
         tool_search: Optional[Union[bool, str, Dict[str, Any], 'ToolSearchConfig']] = False,  # Progressive tool disclosure
         message_steering: Optional[Union[bool, 'MessageSteeringProtocol']] = False,  # Real-time message steering during execution
+        tool_retry_config: Optional['ToolRetryConfig'] = None,  # Automatic retry/backoff for tool failures
```

Ruff (0.15.18)

[error] Undefined name `ToolRetryConfig`

(F821)